src/NBomber/Api/Builder.fs

```diff
@@ -0,0 +1,190 @@
+    member _.Pause(scenario: Scenario, millis : int) =
+        Step.createPause millis |> addTo scenario
+
+    member _.Pause(scenario: Scenario, getMillis : unit -> int) =
+        Step.createPause getMillis |> addTo scenario
+
+    member _.Pause(scenario: Scenario, timeSpan : TimeSpan) =
+        Step.createPause timeSpan |> addTo scenario
+
+    member _.Pause(scenario: Scenario, getTimeSpan : unit -> TimeSpan) =
+        Step.createPause getTimeSpan |> addTo scenario
+
+    /// set warmup duration
+    [<CustomOperation "warmUp">]
+    member _.WarmUp(scenario: Scenario, time) =
+        Scenario.withWarmUpDuration time scenario
+
+    /// run without warmup
+    [<CustomOperation "noWarmUp">]
+    member _.NoWarmUp(scenario: Scenario) =
+        Scenario.withWarmUpDuration TimeSpan.Zero scenario
+
+    /// setup load simulation
+    [<CustomOperation "load">]
+    member _.Load(scenario: Scenario, simulations) =
+        Scenario.withLoadSimulations simulations scenario
+
+    /// run an action before test
+    [<CustomOperation "testInit">]
+    member _.TestInit(scenario: Scenario, init) =
+        Scenario.withTestInit init scenario
+    member _.TestInit(scenario: Scenario, init : ScenarioContext -> Task) =
+        Scenario.withTestInit (unitTask init) scenario
+
+    /// run an ation after test
+    [<CustomOperation "testClean">]
+    member _.TestClean(scenario: Scenario, clean) =
+        Scenario.withTestClean clean scenario
+    member _.TestClean(scenario: Scenario, clean : ScenarioContext -> Task) =
+        Scenario.withTestClean (unitTask clean) scenario
+
+let scenario name = ScenarioBuilder name
+
+/// performance test builder
+type RunnerBuilder() =
+    let empty = NBomberRunner.registerScenarios []
+
+    member _.Zero() = empty
+
+    member _.Yield _ = empty
+
+    member _.Run f = f
+
+    [<CustomOperation "scenarios">]
+    member _.Scenarios(ctx, scenarios) =
+        { ctx with RegisteredScenarios = scenarios }
+
+    [<CustomOperation "reportFileName">]
+    member _.ReportFileName(ctx, fileName) =
+        { ctx with ReportFileName = Some fileName }
+
+    [<CustomOperation "reports">]
+    member _.ReportFormats(ctx, reports) =
+        { ctx with ReportFormats = reports }
+
+    [<CustomOperation "noReports">]
+    member _.NoReports(ctx) =
+        { ctx with ReportFormats = [] }
+
+    [<CustomOperation "testSuite">]
+    member _.TestSuite(ctx : NBomberContext, name) =
+        { ctx with TestSuite = name }
+
+    [<CustomOperation "name">]
+    member _.Name(ctx : NBomberContext, name) =
+        printfn "runner name %s" name
+        { ctx with TestName = name }
+
+    [<CustomOperation "config">]
+    member _.Config(ctx, path) =
+        NBomberRunner.loadConfig path ctx
+
+    [<CustomOperation "configInfrastructure">]
+    member _.ConfigInfrastructure(ctx, path) =
+        NBomberRunner.loadInfraConfig path ctx
+
+    [<CustomOperation "reporter">]
+    member _.Report(ctx, reporter, interval) =
```

Comment:
in NBomber you have 1 interval for all reporting sinks

Reply:
splitted to two keywords: reportInterval and reportSink. Maybe it could also be a ReportBuilder if it were a sub record with reporting properties
```
report {
   interval (seconds 5.0)
   sink mySink
   formats [HTML, CSV]
   fileName "reportFile"
   etc...
}
```

By the way, what is happened to MD format?

Reply:
MD format is temporary removed. 
Reason - requires support on which I don't have a capacity plus TXT covers 99% use cases .

Reply:
report {} looks nice

Reply:
added report configuration. Changed NBomberContext to have a `Repord` property. Doesn't work inplace. Only predefined like here:

```fsharp
let myReporterConfig =
  report {
    fileName "report_file"
    etc ...
  }
let perftest "myTest" {
  report myReporterConfig
  etc ...
}
```